python/lsst/dc2gen3/patch_dc2.py

```diff
@@ -19,70 +23,210 @@
 # constants
 
 DATATYPES_TO_CONVERT = ("flat", "sky", "SKY")
-POLICY_FILE = Policy.defaultPolicyFile(
-    "obs_lsst", "lsstCamMapper.yaml", "policy"
-)
-FIX_HEADER = False
+POLICY_FILE = Policy.defaultPolicyFile("obs_lsst", "lsstCamMapper.yaml", "policy")
 
 # exception classes
 
 # interface functions
 
 
-def copy_files(origin_repo, new_repo, policy_file=POLICY_FILE):
+def move_files(calib_root, policy_file=POLICY_FILE):
     """Copy files from a DESC DC2 gen2 origin repo to one with modified names
     to change parsed filter names to match that expected by the gen3 butler.
 
     Parameters
     ==========
-    origin_repo : str
-        full path to origin gen2 (POSIX) filestore
-    new_repo : str
-        full path to destination (POSIX) filestore
+    calib_root : str
+        full path to the calib gen2 (POSIX) filestore
     policy_file : str
         Policy file from which to load templates
 
     """
     log = lsst.log.Log.getLogger("convertRepo")
     policy = Policy(policy_file)
     for data_type in DATATYPES_TO_CONVERT:
+
         if data_type == "SKY":
             template = policy["calibrations"]["sky"]["template"]
             template = template.replace("sky", "SKY")
         else:
             template = policy["calibrations"][data_type]["template"]
 
-        for old_path, new_path in _transform_pairs(
-            origin_repo, new_repo, template
-        ):
-
-            if FIX_HEADER:
-                assert False, "This path is untested"
-                with fits.open(old_path) as hdulist:
-                    for hdu in hdulist:
-                        if "FILTER" in hdu.header:
-                            old_filter = hdu.header["FILTER"]
-                            new_filter = _transform_filter_name(old_filter)
-                            hdu.header["FILTER"] = new_filter
-                    hdulist.writeto(new_path)
-            else:
-                log.info("Copying %s to %s", old_path, new_path)
-                try:
-                    shutil.copyfile(old_path, new_path)
-                except FileNotFoundError:
-                    new_dir = os.path.split(new_path)[0]
-                    try:
-                        os.makedirs(new_dir)
-                    except FileExistsError:
-                        pass
-                    shutil.copyfile(old_path, new_path)
+        for old_filename, new_filename in _transform_pairs(calib_root, template):
+            log.info("Moving %s to %s", old_filename, new_filename)
+            new_dir = os.path.split(new_filename)[0]
+            try:
+                os.makedirs(new_dir)
+            except FileExistsError:
+                pass
+            shutil.move(old_filename, new_filename)
+
+
+def update_registry(gen2_root):
+    """Update a gen2 registry to include new columns
+
+    Parameters
+    ==========
```

Comment:
Should be single-dash underline (doesn't matter if we're not running Sphinx, but might as well get it right in case it gets copied around).